### PR TITLE
Indicate role using LED eye colors

### LINKED ISSUE #117: Role indication by LED colors
Make the LED in the eyes change color according to the current player role. For example, striker = red, support striker = orange, midfield = white, goalie = blue

### PR DESCRIPTION
**Pull request by Lasse-van-Iterson**
## Introduced Changes

The NAO's will now change their eye color based on their currently assigned role

Fixes #117 

## Ideas for Next Iterations (Not This PR)

- New colors (more easily viewable)

## How to Test

Run code on NAO and force roles using twix

### CODE REVIEW

**Comment by oxkitsune** [comment]
good stuff guys! will take a look

File: crates/control/src/led_status.rs [hunk 1]
```diff
@@ -199,6 +202,16 @@ impl LedStatus {
                 } else {
                     None
                 };
+                let right_eye = match role {
+                    Role::Loser => Eye::loser_eye(),
+                    Role::Striker => Eye::striker_eye(),
+                    Role::StrikerSupporter => Eye::striker_support_eye(),
+                    Role::DefenderLeft | Role::DefenderRight => Eye::defender_eye(),
```
Comment (oxkitsune) [inline crates/control/src/led_status.rs:209]: Could we maybe split this up, and split the colours in the eye based on the defender side? Might be useful!

**Review by oxkitsune — CHANGES_REQUESTED** [review]

File: crates/control/src/led_status.rs [hunk 2]
```diff
@@ -199,6 +202,16 @@ impl LedStatus {
...
+                    Role::Keeper => Eye::keeper_eye(),
+                    Role::ReplacementKeeper => Eye::keeper_replacement_eye(),
```
Comment (Fyor) [inline crates/control/src/led_status.rs:211]:
```suggestion
                    Role::ReplacementKeeper => Eye::replacement_keeper_eye(),
```

File: crates/types/src/led.rs [hunk 3]
```diff
@@ -26,6 +26,99 @@ pub struct Eye {
...
+            color_at_90: Rgb::new(0, 0, 255),
+            color_at_135: Rgb::new(0, 0, 255),
+            color_at_180: Rgb::new(0, 0, 255),
+            color_at_225: Rgb::new(0, 0, 255),
+            color_at_270: Rgb::new(0, 0, 255),
+            color_at_315: Rgb::new(0, 0, 255),
+        }
+    }
+
+    pub fn keeper_replacement_eye() -> Eye {
```
Comment (Fyor) [inline crates/types/src/led.rs:95]:
```suggestion
    pub fn replacement_keeper_eye() -> Eye {
```

**Review by Fyor — CHANGES_REQUESTED** [review]
Looks good, but I would like to have a guide/wiki where I can see the mapping from role to eye color.

File: crates/types/src/led.rs [hunk 4]
```diff
@@ -26,6 +26,99 @@ pub struct Eye {
...
+            color_at_180: Rgb::new(255, 0, 25),
+            color_at_225: Rgb::new(25, 30, 255),
+            color_at_270: Rgb::new(255, 0, 255),
+            color_at_315: Rgb::new(255, 10, 5),
+        }
+    }
+
+    pub fn striker_eye() -> Eye {
+        Eye {
+            color_at_0: Rgb::new(255, 0, 0),
```
Comment (Redjeepkaiser) [inline crates/types/src/led.rs:45]: User Rgb::Red

File: crates/types/src/led.rs [hunk 5]
```diff
@@ -26,6 +26,99 @@ pub struct Eye {
...
+            color_at_180: Rgb::new(255, 255, 0),
+            color_at_225: Rgb::new(255, 255, 0),
+            color_at_270: Rgb::new(255, 255, 0),
+            color_at_315: Rgb::new(255, 255, 0),
+        }
+    }
+
+    pub fn defender_eye() -> Eye {
+        Eye {
+            color_at_0: Rgb::new(255, 255, 255),
```
Comment (Redjeepkaiser) [inline crates/types/src/led.rs:71]: Use rgb::white.

File: crates/types/src/led.rs [hunk 6]
```diff
@@ -26,6 +26,99 @@ pub struct Eye {
...
+            color_at_180: Rgb::new(255, 255, 255),
+            color_at_225: Rgb::new(255, 255, 255),
+            color_at_270: Rgb::new(255, 255, 255),
+            color_at_315: Rgb::new(255, 255, 255),
+        }
+    }
+
+    pub fn keeper_eye() -> Eye {
+        Eye {
+            color_at_0: Rgb::new(0, 0, 255),
```
Comment (Redjeepkaiser) [inline crates/types/src/led.rs:84]: See comments above

**Review by Redjeepkaiser — CHANGES_REQUESTED** [review]
Looks good!

**Review by oxkitsune — APPROVED** [review]

**Pull request merged**